### DISCUSSION TITLE
@lit/localize without TS

### DISCUSSION THREAD

**Discussion by mario-aleo** [lit/lit#2176] [Q&A] [answered]
Do @lit/localize support plain JS projects?

I can't manage to setup @lit/localize on a project that isn't using TS.

Even though https://lit.dev/docs/libraries/localization/ has a **JS** option on exemples, I can run `lit-localize extract` because the needed ` tsconfig.json` can't find any file to work with :(

**Comment by aomarks** [comment, marked as answer]
This should be possible by creating a tsconfig and setting the `allowJs` option: https://www.typescriptlang.org/tsconfig#allowJs

But we should definitely prove this out with an example project and document it at  https://lit.dev/docs/libraries/localization/

I filed https://github.com/lit/lit/issues/2177 and https://github.com/lit/lit.dev/issues/512 to track this work

**Reply by aomarks** [reply]
Follow up: we just released `@lit/localize-tools` v0.4.0 which includes a number of enhancements to make working with vanilla JS much easier, such as not needing to define a `tsconfig.json` at all. See [here](https://github.com/lit/lit/blob/main/packages/localize-tools/CHANGELOG.md#040) for the changes.

The documentation at https://github.com/lit/lit/tree/main/packages/localize has been fully updated with JavaScript examples, so take a look there. We've also added two example projects you can look at for both modes: https://github.com/lit/lit/tree/main/packages/localize/examples/runtime-js and https://github.com/lit/lit/tree/main/packages/localize/examples/transform-js

The https://lit.dev/docs/libraries/localization/ page still hasn't been updated (track at https://github.com/lit/lit.dev/issues/512), but the README actually has the more complete documentation for localize currently anyway (we'll be moving full localize documentation into lit.dev soon).

**Reply by mario-aleo** [reply]
Those are really good news, thank you 😄

**Reply by aomarks** [reply]
The docs at https://lit.dev/docs/localization/overview/ have now also been updated to show JS usage (and are more extensive generally).
